README.md
# ELS - Enterprise Linux/EuroLinux Sources - system-config-network
 
## Usage:
  Checkout branch or tag. Use `make sources` to recreate  SOURCE0 archive.


src/ChangeLog
commit 363593543c504bd83e7974a1cee343fa878f96e2
Author: Harald Hoyer <harald@redhat.com>
Date:   Fri Oct 12 14:48:16 2012 +0200

    fix "make srpm"

commit b8517c7d7fdce6d036a0bfb9a90397713ef4a720
Author: Harald Hoyer <harald@redhat.com>
Date:   Fri Oct 12 14:30:53 2012 +0200

    Add dialogs for TUI activate/deactivate

commit 7509a82901a7558c16ecace975c0dd7c58b1df65
Author: Harald Hoyer <harald@redhat.com>
Date:   Fri Oct 12 13:20:53 2012 +0200

    inc version

commit 1078b9103441c322f3ce5ce01de75616b3ff4943
Author: Harald Hoyer <harald@redhat.com>
Date:   Fri Oct 12 13:21:00 2012 +0200

    Add "Activate" and "Deactivate" buttons to the TUI

commit eb05eb6c0518319f205bdbdb06ada07e3f5936e5
Author: Harald Hoyer <harald@redhat.com>
Date:   Fri Oct 12 13:07:27 2012 +0200

    set NM_CONTROLLED=no for TUI created devices

commit f5c77ec83a9205e26cc543cea3b19291f45f263e
Author: Harald Hoyer <harald@redhat.com>
Date:   Fri Oct 12 13:01:20 2012 +0200

    netconfpkg/NC_functions.py: allow trailing dot for hostnames

commit 282e60fd7309df768d43fe236866026414737d19
Author: Harald Hoyer <harald@redhat.com>
Date:   Fri Oct 12 12:50:08 2012 +0200

    translation update

commit 461804334637d814ecbe65e03fd571a014a46359
Author: jassy <jassy@fedoraproject.org>
Date:   Fri Oct 15 06:37:11 2010 +0000

    l10n: Updates to Panjabi (Punjabi) (pa) translation
    
    Transmitted-via: Transifex (translate.fedoraproject.org)

commit 9078f9bd4a2dc9c70fd0744c056c6579475be9b3
Author: Harald Hoyer <harald@redhat.com>
Date:   Tue Jun 22 18:08:24 2010 +0200

    version 1.6.0.el6.2-1

commit b7fa985f144d1fcbbf5545d9a35434343e302eb1
Author: Jiri Popelka <jpopelka@redhat.com>
Date:   Mon Jun 21 13:33:53 2010 +0200

    Ported to use NSS library (bug #348321).
    
    We use python bindings for NSS/NSPR:
    https://fedoraproject.org/wiki/Features/PythonNSS
    
    If there's no PythonNSS installed hashlib is used instead.

commit e8ac0d9f92c2f7eb2e9fe02dd1dbf58801b081ba
Author: Jiri Popelka <jpopelka@redhat.com>
Date:   Mon Jun 21 11:35:20 2010 +0200

    Use hashlib instead of deprecated sha module.

commit 294d19827b587a02692d5d5984a7b47629ea02e1
Author: Jiri Popelka <jpopelka@redhat.com>
Date:   Thu Jun 17 11:53:47 2010 +0200

    Use hashlib instead of deprecated md5 module.

commit 9b8fcb30d923df7fae24713dfd213bc0da26cfbd
Author: Jiri Popelka <jpopelka@redhat.com>
Date:   Thu Jun 17 11:00:12 2010 +0200

    Fix writing of comments in ConfModules.py

commit 1712d45dfd0dc6e87a6bf61cf0b2ecfcfe94f66e
Author: Jiri Popelka <jpopelka@redhat.com>
Date:   Mon Jun 7 14:05:40 2010 +0200

    Avoid saving of SetVolume = None into wvdial.conf (bug #601005).

commit 05c630d1f23ad5aa03fb763f21a33c04b80e4ba6
Author: Jiri Popelka <jpopelka@redhat.com>
Date:   Mon Feb 8 13:26:19 2010 +0100

    Fix calling of join() when logging (bug #562292)

commit cde1ffe741f054107f6bac81a7bf3f714e4326f4
Author: Jiri Popelka <jpopelka@redhat.com>
Date:   Mon Jan 25 15:03:48 2010 +0100

    Revert "NCPluginDevModem.py:getModemList() requires kudzu"
    
    This reverts commit cc6e29afbcc034d38a3857ef6f58e28ac4ab2b8c.

commit 9a1872ac7f5c41219195a855d855863035a7caf1
Author: Harald Hoyer <harald@redhat.com>
Date:   Wed Jun 16 18:07:45 2010 +0200

    /etc/modprobe.conf is deprecated (#589593).
    
    Read|write /etc/modprobe.d/network.conf instead.

commit 60ecfca687d5ddc2058644a9f22860ff2b5f9887
Author: mgiri <mgiri@fedoraproject.org>
Date:   Tue May 11 10:34:24 2010 +0000

    l10n: Updates to Oriya (or) translation
    
    Transmitted-via: Transifex (translate.fedoraproject.org)

commit 95fb57470dc3257a237643bec787cf89d6a0c362
Author: rajesh <rajesh@fedoraproject.org>
Date:   Wed May 5 11:20:45 2010 +0000

    l10n: Updates to Hindi (hi) translation
    
    Transmitted-via: Transifex (translate.fedoraproject.org)

commit 64b4a371ca02557f1c16efcd528af7aef2e1af8e
Author: enshahar <enshahar@fedoraproject.org>
Date:   Wed Apr 28 03:59:22 2010 +0000

    l10n: Updates to Korean (ko) translation
    
    Transmitted-via: Transifex (translate.fedoraproject.org)

commit 9a7d0e89a2349e9341945bb2bcb08ddb26237f52
Author: shanky <shanky@fedoraproject.org>
Date:   Fri Apr 23 09:05:02 2010 +0000

    l10n: Updates to Kannada (kn) translation
    
    Transmitted-via: Transifex (translate.fedoraproject.org)

commit 96fa4b1af190dea09b60a39ad90d4d117a16e570
Author: khasida <khasida@fedoraproject.org>
Date:   Fri Apr 23 08:11:41 2010 +0000

    l10n: Updates to Japanese (ja) translation
    
    Transmitted-via: Transifex (translate.fedoraproject.org)

commit 90a6ca5727274d269204a4c77c27e654df8ca08e
Author: anipeter <anipeter@fedoraproject.org>
Date:   Wed Apr 14 05:35:52 2010 +0000

    l10n: Updates to Malayalam (ml) translation
    
    Transmitted-via: Transifex (translate.fedoraproject.org)

commit 3a2082461a06f046b69a9ee53645456c6342290c
Author: mvdz <mvdz@fedoraproject.org>
Date:   Tue Apr 13 12:35:08 2010 +0000

    l10n: Updates to Ukrainian (uk) translation
    
    Transmitted-via: Transifex (translate.fedoraproject.org)

commit 6bbe6af7ca46e8c264770427154442fa143a2e6d
Author: gcintra <gcintra@fedoraproject.org>
Date:   Wed Apr 7 03:34:21 2010 +0000

    l10n: Updates to Portuguese (Brazilian) (pt_BR) translation
    
    Transmitted-via: Transifex (translate.fedoraproject.org)

commit 3fbacb2aca1e38e0a48814e4a808306c2914f743
Author: gcintra <gcintra@fedoraproject.org>
Date:   Wed Apr 7 03:34:05 2010 +0000

    l10n: Updates to Portuguese (Brazilian) (pt_BR) translation
    
    Transmitted-via: Transifex (translate.fedoraproject.org)

commit c1ce84f6c05860fd4821c06a87c1985f22cff870
Author: jassy <jassy@fedoraproject.org>
Date:   Tue Apr 6 06:29:22 2010 +0000

    l10n: Updates to Panjabi (Punjabi) (pa) translation
    
    Transmitted-via: Transifex (translate.fedoraproject.org)

commit e41c6c9adea5740a7519805e35712e629d12dce0
Author: gguerrer <gguerrer@fedoraproject.org>
Date:   Mon Mar 29 06:44:34 2010 +0000


src/src/netconfpkg/ipcalc.py
# Python wrapper for ipcalc command line IP address manipulation tool
# Copyright (c) 2001-2005 Red Hat, Inc. All rights reserved.
#
# This software may be freely redistributed under the terms of the GNU
# public license.
#
# You should have received a copy of the GNU General Public License
# along with this program; if not, write to the Free Software
# Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
#
# Author: Preston Brown <pbrown@redhat.com>
import os
import re


class IPCalc:
    """A wrapper class that encapsulates the command line
    functionality of the ipcalc command, providing access to computing
    network prefixes, netmasks, etc. from python."""

    def __init__(self, address, prefix=None, netmask=None):
        """Initialize an IPCalc object.  Address must be provided, and
        in order for most of the other calculations to be possible,
        one of prefix or netmask must also be provided."""
        self._address = address
        self._prefix = prefix
        self._netmask = netmask

    def prefix(self):
        if self._prefix:
            return self._prefix
        else:
            if not self._netmask:
                text = os.popen("ipcalc -m %s" % self._address).read()
                self._netmask = re.match("NETMASK=(\S+)", text).groups()[0]

            text = os.popen("ipcalc -p %s %s" % (
                self._address, self._netmask)).read()
            self._prefix = re.match("PREFIX=(\d+)", text).groups()[0]
            return self._prefix

    def netmask(self):
        if self._netmask:
            return self._netmask
        else:
            if not self._prefix:
                text = os.popen("ipcalc -m %s" % self._address).read()
            else:
                text = os.popen("ipcalc -m %s/%s" % (
                    self._address, self._prefix))
            self._netmask = re.match("NETMASK=(\S+)", text).groups()[0]
            return self._netmask

    def network(self):
        if not self._prefix:
            self.prefix()
        text = os.popen("ipcalc -n %s/%s" % (self._address, 
                                             self._prefix)).read()
        return re.match("NETWORK=(\S+)", text).groups()[0]

    def broadcast(self):
        if not self._prefix:
            self.prefix()
        text = os.popen("ipcalc -b %s/%s" % (self._address, 
                                             self._prefix)).read()
        return re.match("BROADCAST=(\S+)", text).groups()[0]

def test():
    """Test function for the IPCalc class."""

    ipc = IPCalc("207.175.42.15", netmask="255.255.254.0")
    print "prefix:", ipc.prefix()
    print "netmask:", ipc.netmask()
    print "bcast:", ipc.broadcast()
    print "network:", ipc.network()


if __name__ == "__main__":
    test()


__author__ = "Harald Hoyer <harald@redhat.com>"
__date__ = "$Date: 2007/03/14 09:29:37 $"
__version__ = "$Revision: 1.8 $"
